P/p3a/build_tarballs.jl

```diff
@@ -0,0 +1,68 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "p3a"
+version = v"1.0.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/sandialabs/p3a.git", "271cd1d77fbc850aa4ddfb015b59176926670b3b")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/p3a/
+mkdir build && cd build
+
+CMAKE_FLAGS=()
+
+#help find mpi on mingw, subset of https://github.com/JuliaPackaging/Yggdrasil/blob/b4fdb545c3954cff218051d7520c7418991d3416/T/TauDEM/build_tarballs.jl#L28-L53
+if [[ "$target" == x86_64-w64-mingw32 ]]; then
+    CMAKE_FLAGS+=(
+        -DMPI_HOME=${prefix}
+        -DMPI_GUESS_LIBRARY_NAME=MSMPI
+    )
+    if [[ "${target}" == x86_64-* ]]; then
+        for lang in C CXX; do
+            CMAKE_FLAGS+=(-DMPI_${lang}_LIBRARIES=msmpi64)
+        done
+    fi
+fi
+
+cmake .. \
+-DCMAKE_INSTALL_PREFIX=$prefix \
+-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+-DCMAKE_BUILD_TYPE=Release \
+-DBUILD_TESTING=OFF \
+-DBUILD_SHARED_LIBS=ON \
+-DKokkos_COMPILE_LANGUAGE=CXX \
+"${CMAKE_FLAGS[@]}"
+
+make -j${nproc}
+make install
+
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms(; experimental = true))
+#Kokkos is only available on 64 bit
+filter!(p -> nbits(p) != 32, platforms)
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libp3a", :libp3a)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="Kokkos_jll", uuid="c1216c3d-6bb3-5a2b-bbbf-529b35eba709"))
+    Dependency(PackageSpec(name="MPICH_jll", uuid="7cb0a576-ebde-5e09-9194-50597f1243b4"))
```

Comment:
See https://github.com/JuliaPackaging/Yggdrasil/blob/2fc2e9a55180b90ab96f20e582bdde2913c6a940/A/ADIOS2/build_tarballs.jl#L122-L123
to use the MPI packages only where they're available